tools/obscuroscan/obscuroscan.go

```diff
@@ -138,7 +140,39 @@ func (o *Obscuroscan) getNumTransactions(resp http.ResponseWriter, _ *http.Reque
 	_, err = resp.Write([]byte(numTransactions.String()))
 	if err != nil {
 		log.Error("could not return total number of transactions to client. Cause: %s", err)
-		logAndSendErr(resp, "Could not fetch number of rollups.")
+		logAndSendErr(resp, "Could not fetch total transactions.")
+		return
+	}
+}
+
+// Retrieves the average rollup time, as (time last rollup - time first rollup)/number of rollups
+func (o *Obscuroscan) getRollupTime(resp http.ResponseWriter, _ *http.Request) {
+	numLatestRollup, err := o.getLatestRollupNumber()
+	if err != nil {
+		log.Error(err.Error())
+		logAndSendErr(resp, "Could not fetch average rollup time.")
+		return
+	}
+
+	firstRollup, err := o.getRollupByNumber(0)
```

Comment:
Worth keeping `startTime := firstRollup.Header.Time` in memory on the Obscuroscan state instead of fetching it on every request?